### ISSUE TITLE
Applying on a trained model

### ISSUE THREAD

**Issue by mwdotzom** [XiangMochu/MDEUncertainty#2]
Hi Mochu, thanks again for the great work!

In your paper, I noticed you studied [GrUMoDepth](https://github.com/jhornauer/GrUMoDepth) in §1.Introduction, and examined combined result with your design in §4.7.Discussion. These are some vague thoughts I have towards the two:

1. As a _post-hoc_ method, GrUMoDepth does not require training a new model - it utilizes flipping + gray-scaling + noising + rotation of inputs **after the model's inference**. But as a result, the uncertainty map is less aligned due to limited sources.

2. On the contrast, MDEUncertainty introduced a new ranking-based regularizer, which boosts uncertainty quality. However, it is guided through loss **in training**, and generated **as an extra metric in the model's inference**. Therefore, it is integrated and has to be trained in a new model.

3. [MiDaS](https://github.com/isl-org/MiDaS) has high resistance to noise thanks to **huge training data**. It's preferable to make use of both its large training and MDEUncertainty. However, MiDaS does not plan to release training code. Therefore, I would have to either use two models separately, or give up the idea and train from scratch.

Can you share your comments or perhaps point out what is inaccurate? This would be utterly helpful for understanding your work!

Thank you!! Haokun

**Comment by XiangMochu** [comment]
Hi, here are some of my understandings:

1) Flip augmentation, for example, should be invariant to model predictions, but the model provides slightly different predictions, and it can reveal uncertainty to some extent. GrUMoDepth back-propagate this different and get intermidiate gradient maps in latent features, and find it to be more correlated with the oracle uncertainty (the actual error).

2) Our method, is a training-based one, which requires training a model to provide decent uncertainty representations.

3) If you want to derive uncertainty from MiDaS, you can try: simply measure the difference between flipped inputs / use GruMoDepth to extract gradients of the latent features. However, MiDaS does not provide metric depth, if you want to adapt it to give metric depth predictions you will need to train it anyway.

**Comment by mwdotzom** [comment]
Thank you so much! This helps a lot, I'll reopen the issue if I have more questions on this topic :D
